doc/index.md

```diff
@@ -143,12 +137,12 @@ Checks the socket for a valid Transport Layer Security connection. Returns the l
 
 ### `connection:localname()` <!-- --> {#connection:localname}
 
-Returns the connection information for the local socket. Returns address family, IP address and port for an external socket. For Unix domain sockets, the function returns `AF_UNIX` and the path. If the connection object is not connected, returns `AF_UNSPEC` (0). 
+Returns the connection information for the local socket. Returns address family, IP address and port for an external socket. For Unix domain sockets, the function returns `AF_UNIX` and the path. If the connection object is not connected, returns `AF_UNSPEC` (0). If the error is other than a connection error, returns `nil` an error message and an error number.
```

Comment:
> If the error is other than a connection error

I don't like this phrasing, e.g. what is a "connection error"?

It should be enough to just say "on error, returns ..."